v2/client.go

```diff
@@ -485,6 +494,20 @@ func (r *Response) Error() error {
 	return nil
 }
 
+// CSVResponse represents a list of statement results in CSV form.
+type CSVResponse struct {
```

Comment:
I have two thoughts on this. I think this can go in one of two directions.

1. Maybe `QueryAsCSV` should return a struct that has the same API as [csv.Reader](https://godoc.org/encoding/csv#Reader). It can likely be like this:

```go
type CsvReader struct {
    reader *csv.Reader
    close func() error
}
```

2. You can also expose a `QueryAsRaw` that allows specifying the mime type as an argument and have it return an `io.ReadCloser` if you just want direct access to the `io.ReadCloser`.

Reply:
Agreed, but I do think the second one is more useful. If you need to parse the CSV you can later feed it to the `csv.Reader`.

On the other hand, if you just need to export data to a file, for example, and from a glance of that package documentation, it seems you would need to read all the records in the CSV and build it again just to write to the file.

What do you think?

Reply:
I think the `io.ReadCloser` would be more useful to expose, but I also worry that the API is easy to get wrong. If it is the raw data you need, the `net/http` API is a lot better to use than the InfluxDB client library.

I'm wondering if this use case would be better served by just using `net/http` rather than building it into the v1 client API. What are your thoughts on this?

The influxdb client library is really only useful for encoding points to line protocol and decoding JSON to structs. If you are reading to csv, then the tools in the standard library will probably serve you better.

Reply:
I would honestly expect that, if the InfluxDB server API allows for CSV queries, it would be implemented in the official library. I was surprised to learn it doesn't.

So my opinion is that it should be implemented. Otherwise third parties using this library would have to duplicate code for handling things like authentication on their end to use `net/http` just for making the CSV queries.

Nonetheless this is just my opinion and you should weight all of the pros and cons since balancing an API is always difficult and you're far more qualified to understand your clients needs than I am.

Reply:
It might have been more helpful for the API to integrate more with `net/http` but I think that's the plan for the API that interacts with v2. Trying to add it on to the current client API seems like a bigger task than we would be willing to undertake so, for this, I think it might be reasonable to have something like the following:

```go
type QueryOption interface {
    // unexported method used for customizing raw queries
}

type Client interface {
    QueryAsRaw(q Query, opts ...QueryOption) (io.ReadCloser, http.Header, error)
}
```

If you're willing to fill in the implementations, I can write the stubs for the API and push them into this branch.

@rawkode any thoughts?

Reply:
These changes and another for the server response decode are all it's needed to support this feature, BTW.

Nevertheless I'm willing to fill the implementations.

Reply:
It occurred to me late Friday afternoon that any change to the interface may be out of scope. A change to the interface means that the API has a breaking change for anybody who may add something like a mock implementation outside of the scope of this library.

I think we can do something like adding a free function like `MakeRequest` to aid in the creation of an `http.Request` object, but modifying the interface may be out of our ability.

Adding @dgnorton as a cc on this.

Reply:
Are there any news regarding this issue? @jsternberg @rawkode @dgnorton